### DISCUSSION TITLE
Which do you prefer, the old UI or the new V4 UI ?

### DISCUSSION THREAD

**Discussion by GaiheiluKamei** [activeadmin/activeadmin#8229] [Polls]
as described.

**Comment by caiotarifa** [comment]
I'm really happy with the new interface! But, it would be great if ActiveAdmin could lean towards a 'headless' style. This means focusing more on streamlining the backend and cutting down on CSS and JS to a bare minimum. It would make customizing things a lot simpler and more efficient.

**Reply by caiotarifa** [reply]
Maybe having a separated Gem with the default theme.

**Reply by javierjulio** [reply]
To be honest I think of it that way already. It’s meant to be a base style that you can easily modify. More so than before. Selectors are written with low specificity which helps greatly.

**Reply by GaiheiluKamei** [reply]
Maybe it's because the old style has been around for a long time and I'm not used to the new changes. The major adjustment for me was primarily caused by significant alterations in fonts, layouts, and other elements, resulting in a notably reduced information density per page. 

However, I truly value the diligent efforts of the maintainers, I just need some time to adapt, and I'm also curious to know if others are feeling the same way.

**Reply by javierjulio** [reply]
@GaiheiluKamei thank you. I really appreciate that clarification as it comes across otherwise. A significant amount of time and effort has gone into this release, without support, in an effort to give the library a new lease of life for many more years.

The theme is just a default for light and dark modes. It's intentional to be clean and simple. Not only can you, you are meant to customize it. No one is forcing this default theme and/or layout on you. The extraction to partials was done so you could modify the styles and layout easily and ideally without monkey patches. For example, I've seen gems for changing the sidebar position. It's now [in the layout file in 2 lines wrapped by a simple CSS grid](https://github.com/activeadmin/activeadmin/blob/master/app/views/layouts/active_admin.html.erb#L13-L16). No need to monkey patch anything. You have the ability to make design preference changes yourself through simpler CSS overrides than before. That's very much by design here.

**Reply by javierjulio** [reply]
@caiotarifa that's an interesting point of view. In my experience, others prefer something ready out of the box. Using another gem to provide styles though has its own issues. I wouldn't be for that primarily because of the issues around Rails engines that provide frontend code and assets. There isn't much support around that. Others have high expectations now e.g. no-node/no-build or all node. There isn't anymore plug-in-play type of integration either. Hence the approach we are taking with v4.

**Reply by GaiheiluKamei** [reply]
I have been exploring various CMS platforms in different languages for some time, and I have developed a newfound appreciation for ActiveAdmin. I found that even though Strapi is quite good, it lacks the simplicity, elegance, and customizability that ActiveAdmin offers. Thank you so much for all that you've done for the Rails community!

**Reply by julioag-rmd** [reply]
I love the new UI, but imposing tailwind as a dependency for Rails projects is kinda bad.

Imagine I have a project that serves some User facing content and uses AA as the backoffice. I'm using Tailwind 3 for all styling.

Now, if I want to move from Tw3 to 4 I need for AA to release a version with that change, so in my TW v4 upgrade I need also to upgrade AA. If I have customizations in the AA views or my own views, that upgrade alone is big enough.

This makes the scope of the work bigger, riskier and harder to tackle.

I know that making decisions that make AA center of a project and indirectly dictate your upgrade pace is bad, but here we are 😅

**Reply by tagliala** [reply]
> but imposing tailwind as a dependency for Rails projects is kinda bad.

Fortunately, Tailwind is not a requirement for ActiveAdmin. In fact, ActiveAdmin does not depend on cssbundling, tailwindcss-rails, or tailwindcss-ruby.

<img width="356" height="206" alt="image" src="https://github.com/user-attachments/assets/dbc9c8f1-c323-475a-af16-e9a608c1d570" />

As a side note, I’d like to mention that `kaminari` is **imposed**. `kaminari` extends all ActiveRecord classes at the application level with pagination functionality, which is applied even outside of ActiveAdmin. This approach can cause conflicts if you prefer to use other pagination libraries, such as pagy.

Returning to the topic of Tailwind, thanks to @jaynetics, you can use [activeadmin_assets](https://github.com/jaynetics/activeadmin_assets) to serve ActiveAdmin assets without needing Tailwind.

**Reply by julioag-rmd** [reply]
Thanks for the response @tagliala. From the upgrade guide for beta 15, I understood I needed to have tailwind in my gemfile to make AA4 work. Re-reading, it seems not, but still need to deal a bit with tailwind to build the css. My bad.

Question: even with Tailwind not being a hard requirement, if I am on version N of Tailwind, which is the same as ActiveAdmin at some point, and I want to go to tailwind N+1, will it be OK or will I need to wait for AA to catch up?

Thanks.

**Reply by tagliala** [reply]
> will it be OK or will I need to wait for AA to catch up? 

I think it could be an issue if the new Tailwind version is not backward compatible, especially because of the dependency on flowbite. While it might not be a blocking problem, it could still cause difficulties in upgrading.

If you do not need to customize Active Admin, such as using different Tailwind classes in partials or creating new components, it may be a good idea to consider https://github.com/jaynetics/activeadmin_assets for serving pre-compiled assets.

Also, we're a bit off topic here since this is a poll about the UI. For a more relevant discussion, you might want to look at https://github.com/activeadmin/activeadmin/discussions/8223.

**Reply by julioag-rmd** [reply]
Thank you!
